homeassistant/components/isy994/switch.py

```diff
@@ -87,3 +120,62 @@ async def async_turn_off(self, **kwargs: Any) -> None:
     def icon(self) -> str:
         """Get the icon for programs."""
         return "mdi:script-text-outline"  # Matches isy program icon
+
+
+class ISYEnableSwitchEntity(ISYAuxControlEntity, SwitchEntity):
+    """A representation of an ISY program switch."""
+
+    def __init__(
+        self,
+        node: Node,
+        control: str,
+        unique_id: str,
+        description: EntityDescription,
+        device_info: DeviceInfo | None,
+    ) -> None:
+        """Initialize the ISY Aux Control Number entity."""
+        super().__init__(
+            node=node,
+            control=control,
+            unique_id=unique_id,
+            description=description,
+            device_info=device_info,
+        )
+        self._attr_name = description.name  # Override super
+        self._change_handler: EventListener = None
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to the node control change events."""
+        self._change_handler = self._node.isy.nodes.status_events.subscribe(
+            self.async_on_update,
+            event_filter={
+                TAG_ADDRESS: self._node.address,
+                ATTR_ACTION: NC_NODE_ENABLED,
+            },
+            key=self.unique_id,
+        )
+
+    @callback
+    def async_on_update(self, event: NodeChangedEvent, key: str) -> None:
+        """Handle a control event from the ISY Node."""
+        self.async_write_ha_state()
+
+    @property
+    def available(self) -> bool:
+        """Return entity availability."""
+        return True  # Enable switch is always available
```

Comment:
Should this check if the ISY is connected?

Reply:
Yes, but are you opposed to that being a follow up PR or do you want to see it all together? Just concious of adding more into this one after I took another look at it. I'm also going to split out the enable/disable switch.

Reply:
> Yes, but are you opposed to that being a follow up PR or do you want to see it all together?

Followup PR is fine for this.  I'd only ask for it in the same PR for first time contributors since I don't know if they are going to ghost after the PR :)

Reply:
I will pick this up for the events update coming up. Right now `isy.connected==True` really only tells you if the module has been initialized, and the config entry reload already makes the entities unavailable. 

The websocket has an `EventsEmitter` that can be used to push the status changes, but I'd like to centralize some of that logic on the HA side and test it more to make sure we don't end up with what's going on with Unifi where the devices go unavailable all the time when the WS resets.